### ISSUE TITLE
s_sha256_compress from tomcrypt is very slow

### ISSUE THREAD

**Issue by saagarjha** [Vector35/binaryninja-api#3593]
**Version and Platform (required):**
- Binary Ninja Version: 3.2.3801 Personal (Build ID 947d78b4)
- OS: macOS
- OS Version: 13.0 (22A379)
- CPU Architecture: Apple silicon

**Bug Description:**
Binary Ninja appears to use SHA-256 hashes as part of its update process. On my machine it spends 10 seconds in what I take to be `s_sha256_compress`, which means I wait a pretty long time for the app to launch. The SHA-256 code on macOS is hardware accelerated and runs at 2+ GB/s, rather than the <200 MB/s that tomcrypt's implementation provides. It would be nice if you used it, or something with similar performance.

**Steps To Reproduce:**
Please provide all steps required to reproduce the behavior:
1. Update Binary Ninja.
2. Wait for it to launch.

**Expected Behavior:**
The app should launch quickly.

**Screenshots:**
<img width="499" alt="image" src="https://user-images.githubusercontent.com/13786931/197392265-cf6659aa-6177-4eb2-8e79-f64c7d1990bc.png">

**Comment by CouleeApps** [comment]
Can confirm that is sha256_compress you're looking at. A quick test with CC_SHA256 saw a decent speed speed up (~3-5x but I didn't measure too closely) on my M1 machine, so this is certainly a big improvement. The one factor that will hold this back is testing the update code and making sure nothing breaks, so it will be a bit before any of this is shipped. Good find on the performance testing though, thanks for the analysis!

**Comment by D0ntPanic** [comment]
tomcrypt has been replaced with botan, which should be much faster

**Comment by saagarjha** [comment]
I see similar performance in 3.3.3997-dev Personal (Build ID 6815c946) (well, I guess the build before that one?). The code looks basically the same, too. Has this shipped?

**Comment by psifertex** [comment]
Yes, we should be using botan in that build. We'll double check and confirm.

**Comment by psifertex** [comment]
Re-opening this issue as we need to do some verification that the performance is actually better now that we've swapped out libraries. We believe that the recent defender performance issues might be exacerbated by this still not being optimized.

**Comment by saagarjha** [comment]
Looks fixed to me, thanks
